### PR TITLE
[Feature] Display active project in navigation drawer

### LINKED ISSUE #230: [Active project] Indicate currently active project
After selecting a project from the side-drawer, the title of the selected project is not shown anywhere in the app. 

I propose that, we add that info somewhere in the app or change the UI to make it more intuitive (For e.g. Instead of opening a dialog when selecting "Select Project", we expand it and highlight the one selected)

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Closes #230  

- [x]  Highlight selected project
- [x]  Clicking on any project activates it
- [x]  Add `+ Join Project` item
- [x]  Display only joined projects in nav drawer

Screenshot:
image

### CODE REVIEW

**Comment by shobhitagarwal1612** [comment]
@gino-m Ready for your review. I have made the changes we discussed on hangouts yesterday

File: gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenFragment.java [hunk 1]
```diff
@@ -424,7 +427,7 @@ public boolean onBack() {
   @Override
   public boolean onNavigationItemSelected(@NonNull MenuItem item) {
     if (item.getGroupId() == R.id.group_join_project) {
-      projectSelectorViewModel.activateProject(item.getOrder());
+      projectSelectorViewModel.activateLocalProject(item.getOrder());
```
Comment (gino-m) [inline gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenFragment.java]: We call these "local" here, and "saved" above.  Is there a more consistent name we can use? From the user's perspective, they will "join/leave" a project, which will store them locally for offline use, and sync them when online. Maybe "offline projects" would be clear?

File: gnd/src/main/java/com/google/android/gnd/repository/ProjectRepository.java [hunk 2]
```diff
@@ -131,6 +131,10 @@ public void activateProject(String projectId) {
         .compose(Loadable::loadingOnceAndWrap);
   }
 
+  public Flowable<Loadable<List<Project>>> getSavedProjects() {
```
Comment (gino-m) [inline gnd/src/main/java/com/google/android/gnd/repository/ProjectRepository.java]: `getOfflineProjectsOnceAndStream()`, as per https://github.com/google/ground-android/wiki/RxJava-Usage-Guidelines#observable-and-flowable

**Comment by shobhitagarwal1612** [comment]
@gino-m Please take a look now!

**Comment by shobhitagarwal1612** [comment]
Gentle ping...

**Comment by shobhitagarwal1612** [comment]
@gino-m Thanks for your comments!

> When I click "+ Join project", the newly joined project doesn't show up under "Projects". Any idea what could be going wrong?

Fixed.

> Also, note that the "Projects" heading in the menu appears above "Offline areas" and "Sign out" which is obviously incorrect. Could we add a separator above Offline areas to make it clear it's a different section, and perhaps change the heading to say "My projects" to make it clear it's not an exhaustive list of available projects?

Fixed

image

File: gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenFragment.java [hunk 3]
```diff
@@ -210,6 +204,13 @@ private void addProjectToNavDrawer(List<Project> projects) {
           .add(R.id.group_join_project, Menu.NONE, index, projects.get(index).getTitle())
           .setIcon(R.drawable.ic_menu_project);
     }
+
+    // Highlight active project
+    Loadable<Project> activeProject = viewModel.getActiveProject().getValue();
+    if (activeProject != null) {
+      int index = getSelectedProjectIndex(activeProject.value().get());
```
Comment (gino-m) [inline gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenFragment.java]: If the project is not yet activated, `value()` will return null here. Instead, you can replace the  `activeProject` null check with `Loadable.getValue(activeProject)` to cover both cases (i.e. Loadable.getValue(activeProject).ifPresent(p => updateSelectedProjectUI(getSelectedProjectIndex(p)));

**Pull request merged**